src/kilo.c

```diff
@@ -939,13 +939,45 @@ void editorMoveCursorToRowEnd() {
     }
     int rowlen = row ? row->size : 0;
     int size = rowlen - E.cx-E.coloff;
-    
-    while (size != 0 ) {
+
+    while (size != 0) {
         editorMoveCursor(RIGHT);
         size = size -1;
     }
 }
 
+int editorMoveCursorToFirst(char c) {
+  int filerow = E.rowoff+E.cy;
+  erow *row = (filerow >= E.numrows) ? NULL : &E.row[filerow];
+
+  if (row == NULL) {
+    return 0;
+  }
+
+  int rowlen = row ? row->size : 0;
+  int size = rowlen;
+
+  int i = E.cx-E.coloff;
```

Comment:
Don't you mean E.cx+E.coloff? We want to start searching at the cursor's position in the file. BTW, once you rebase, just use cursorX() for this.